src/services/tx/tx-sender/dispatch.ts

```diff
@@ -266,6 +270,8 @@ export const dispatchSpendingLimitTxExecution = async (
   return result?.hash
 }
 
-export const dispatchSafeAppsTx = (txId: string, safeAppRequestId: RequestId) => {
-  txDispatch(TxEvent.SAFE_APPS_REQUEST, { txId, safeAppRequestId })
+export const dispatchSafeAppsTx = async (safeTx: SafeTransaction, safeAppRequestId: RequestId) => {
+  const sdk = getAndValidateSafeSDK()
+  const safeTxHash = await sdk.getTransactionHash(safeTx)
```

Comment:
These can both throw.

Reply:
Good catch (pun not intended).

Reply:
Wrapped the parent function in a try-catch.